### PR TITLE
feat: cassandra ssl and auth

### PR DESCRIPTION
**Pull request by TheSriram**
- Add ssl and authentication options for the cassandra driver

### CODE REVIEW

**Comment by racklanta** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/222/

File: deuce/drivers/cassandra/cassandrametadatadriver.py [hunk 1]
```diff
@@ -255,8 +260,26 @@ def __init__(self):
...
             '{0}.cluster'.format(conf.metadata_driver.cassandra.db_module))
 
+        # Import the auth submodule
+        auth_module = importlib.import_module(
+            '{0}.auth'.format(conf.metadata_driver.cassandra.db_module))
+
+        if conf.metadata_driver.cassandra.ssl_enabled:
+            ssl_options = {
+                'ca_certs': conf.metadata_driver.cassandra.ssl_ca_certs,
+                'ssl_version': ssl.PROTOCOL_TLSv1
```
Comment (BenjamenMeyer) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: Is there a way to make this TLS v1.2?

Comment (TheSriram) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: Lets make it a config option.

**Comment by BenjamenMeyer** [comment]
:+1:

**Comment by racklanta** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/229/

**Comment by racklanta** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/232/

**Comment by racklanta** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/238/

**Comment by racklanta** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/241/

File: deuce/drivers/cassandra/cassandrametadatadriver.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import importlib
 import datetime
 import six
+import ssl
```
Comment (painterjd) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: Looks like an unused import?

**Comment by painterjd** [comment]
I only had the one nit-pick regarding the unused import. Once resolved, :+1:

File: deuce/drivers/cassandra/cassandrametadatadriver.py (hunk 2 above)
Comment (TheSriram) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: https://github.com/rackerlabs/deuce/pull/242/files#diff-a8433b86044f7e45283c2cd1d0c2cf6bR268

we need it to get the TLS version.

Comment (BenjamenMeyer) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: @TheSriram is that from before we moved the TLS version to the config?

Comment (TheSriram) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: Nope, after. we do a 
```python
getattr(ssl,'Required TLS version')
```

Comment (painterjd) [inline deuce/drivers/cassandra/cassandrametadatadriver.py]: Ah, missed that.

**Pull request merged**